server/app/services/cloud/rpc_server.rb

```diff
@@ -7,12 +7,12 @@ class RpcServer
     }
 
     class Error < StandardError
-      attr_accessor :code, :message, :backtrace
+      attr_reader :code
 
-      def initialize(code, message, backtrace = nil)
-        self.code = code
-        self.message = message
-        self.backtrace = backtrace
+      def initialize(code, message, remote_backtrace = nil)
+        @code = code
+        super(message)
+        set_backtrace(Array(backtrace) + (['<Remote backtrace>'] + Array(remote_backtrace))) if remote_backtrace
```

Comment:
I don't know how you would raise such an error with a local backtrace.

```
raise Error, "foo"
#and
raise Error, 10, "Foo"
#and
raise Error, 10, "Foo", ["dang"]
```

will give you ArgumentError because of initialize expecting 2+ arguments

```
raise Error.new(10, 'foo', ['dang'])
```
will give you no local backtrace.